masschallenge/impact-api
# [AC-6217] Allocation analysis should not break if option_spec has `option` field set

https://masschallenge.atlassian.net/browse/AC-6217

The deleted lines from `OptionAnalysis.find_options` lines short-circuit matching criterion lookup if the option_spec has an option value set, causing dict lookup failures.

Note that the deleted lines are the default behavior for `CriteriaHelper.options`. The MatchingCriteriaHelpers override this behavior, and the ReadsCriterion and the various JudgeCriteria do not override it, so we get this behavior through the Helper method. 

To test:
on clean db, Boston 2017 Round 1 has criteria set up. in the shell, the following command will create the state that should fail on development and succeed on the feature branch:
```
mc.CriterionOptionSpec.objects.filter(criterion__type="matching").update(option="asdfasdf")
```

No other behavior should change.
